pennylane/ops/op_math/decompositions/unitary_decompositions.py

```diff
@@ -561,149 +480,257 @@ def _decompose_1_cnot(U, wires, initial_phase):
     return math.cast_like(-np.pi / 4, initial_phase)
 
 
-def _decompose_2_cnots(U, wires, initial_phase):
-    r"""If 2 CNOTs are required, we can write the circuit as
-     -╭U- = -A--╭X--RZ(d)--╭X--C-
-     -╰U- = -B--╰C--RX(p)--╰C--D-
-    We need to find the angles for the Z and X rotations such that the inner
-    part has the same spectrum as U, and then we can recover A, B, C, D.
-    """
+def _multidot(*matrices):
+    mat = matrices[0]
+    for m in matrices[1:]:
+        mat = math.dot(mat, m)
+    return mat
+
 
-    # Compute the rotation angles
-    u = math.dot(E_dag, math.dot(U, E))
-    gammaU = math.dot(u, math.T(u))
-    evs = math.linalg.eigvals(gammaU)
+def _real_imag_split_eigh(A, factor):
+    _, basis = math.linalg.eigh(math.real(A) / factor + factor * math.imag(A))
+    eigvals = _multidot(math.transpose(basis), A, basis)
+    return eigvals, basis
 
-    # These choices are based on Proposition III.3 of
-    # https://arxiv.org/abs/quant-ph/0308045
 
-    x = math.angle(evs[0])
+def _ai_kak(U):
+    """Compute a type-AI Cartan decomposition of ``U`` in the standard basis/representation.
+    This is done in the following steps (see App.E of https://arxiv.org/abs/2503.19014, case AI):
+    - compute a real-valued eigenbasis O_1 (orthogonal matrix) and the eigenvalues d^2 of Δ = U U^T,
+    - make O_1 *special* orthogonal by adjusting the sign of its first column,
+    - take the square root of d^2 to obtain d,
+    - compute O_2 = d* O_1^T U,
+    - adjust the sign of the first row of O_2 so that O_2 is special orthogonal, and multiply
+      the first entry of d with the same sign.
+    """
 
-    # If it was the conjugate, grab a different eigenvalue.
-    y = math.cond(
-        math.allclose(evs[0], math.conj(evs[1])),
-        lambda: math.angle(evs[2]),
-        lambda: math.angle(evs[1]),
+    Delta = math.dot(U, math.transpose(U))
+    # Delta^T=Delta implies that Delta, real(Delta), and imag(Delta) share an eigenbasis
```

Comment:
```suggestion
    # U is unitary, so Delta = U U^T is also unitary (product of unitaries).
    # Since Delta is unitary, it is normal (commutes with its adjoint: Delta Delta† = Delta† Delta).
    # For a normal matrix, real(Delta) and imag(Delta) commute with each other and with Delta,
    # therefore they share a common eigenbasis.
```

Reply:
I made this argument more specific to the unitary case, hope that's okay, too.